resources/prosody-plugins/mod_muc_user_hidden.lua

```diff
@@ -0,0 +1,36 @@
+--This module adds is_hidden tag when jwt contains `hidden`
+--flag set to true, or removes it in case it was
+--added maliciously from client sent presence.
+--The module must be enabled under the muc component.
+local is_user_hidden = module:require "util".is_user_hidden;
+local tag_name = "is_hidden"
+
+module:log("info", "Loading mod_muc_user_hidden!");
+
+function add_hidden_tag(event)
+    local stanza = event.stanza;
+    local session = event.origin;
+
+    if stanza == nil or stanza.name ~= "presence" then
+      return
+    end
+
+    stanza:maptags(function(tag)
+      if tag and tag.name == tag_name then

```

Comment:
Why do we need to do that on every presence? 
I think this will be problematic in big meetings as this will be executed on every presence that pass through prosody.

Reply:
Answer to myself: to make sure nobody else is using it ... yeah this worries me that will have a high impact on prosody cpu usage, especially on setting up a big meeting.

Reply:
Nope ignore the session part.

Maybe we can add a feature to the features advertised and the client activates all the logic here only if this is enabled, otherwise even with the tag in presence it will not work.

Reply:
Even better that flag can be used only in lib-jitsi-meet. And if the lib is not adding the isHidden nothing from the logic works.

Reply:
you mean adding the is_hidden on self presence? that would mean parsing the JWT client side...